### DISCUSSION TITLE
python publisher does not communicate with C subscriber

### DISCUSSION THREAD

**Discussion by pdematteis** [eclipse-zenoh/roadmap#133] [Q&A]
I'm trying to have a zenoh python publisher communicating with a zenoh-pico subscriber in peer-to-peer but this is not working.
Scouting seems to send different messages (22 bytes in C, 3 bytes in python).
I'd like to have it working in both TCP and UDP multicast.
Is that supposed to work?

**Comment by Wjun-1** [comment]
> 我正在尝试让 zenoh python 发布者与点对点的 zenoh-pico 订阅者进行通信，但这不起作用。侦查似乎发送了不同的消息（C 语言为 22 字节，Python 语言为 3 字节）。我希望它同时在 TCP 和 UDP 多播中工作。这应该有效吗？

hi,Have you achieved it, please

**Comment by Wjun-1** [comment]
> I'm trying to have a zenoh python publisher communicating with a zenoh-pico subscriber in peer-to-peer but this is not working. Scouting seems to send different messages (22 bytes in C, 3 bytes in python). I'd like to have it working in both TCP and UDP multicast. Is that supposed to work?
Excuse me again, can your zenoh-python and zenoh-pico be subscribed and published? Thank you very much for your reply
